system/lib/gl.c

```diff
@@ -2559,10 +2382,193 @@ for line in open('a').readlines():
   if (!strcmp(name, "glUniformBlockBinding")) return emscripten_glUniformBlockBinding;
   if (!strcmp(name, "glReleaseShaderCompiler")) return emscripten_glReleaseShaderCompiler;
   if (!strcmp(name, "glShaderBinary")) return emscripten_glShaderBinary;
-  if (!strcmp(name, "glGetShaderPrecisionFormat")) return emscripten_glGetShaderPrecisionFormat;
   if (!strcmp(name, "glDepthRangef")) return emscripten_glDepthRangef;
   if (!strcmp(name, "glClearDepthf")) return emscripten_glClearDepthf;
   if (!strcmp(name, "glVertexAttribDivisor")) return emscripten_glVertexAttribDivisor;
+  if (!strcmp(name, "glGenFramebuffers")) return emscripten_glGenFramebuffers;
+  if (!strcmp(name, "glGenRenderbuffers")) return emscripten_glGenRenderbuffers;
+  if (!strcmp(name, "glBindFramebuffer")) return emscripten_glBindFramebuffer;
+  if (!strcmp(name, "glBindRenderbuffer")) return emscripten_glBindRenderbuffer;
+  if (!strcmp(name, "glGetRenderbufferParameteriv")) return emscripten_glGetRenderbufferParameteriv;
+  if (!strcmp(name, "glFramebufferRenderbuffer")) return emscripten_glFramebufferRenderbuffer;
+  if (!strcmp(name, "glRenderbufferStorage")) return emscripten_glRenderbufferStorage;
+  if (!strcmp(name, "glCheckFramebufferStatus")) return emscripten_glCheckFramebufferStatus;
+  if (!strcmp(name, "glDeleteFramebuffers")) return emscripten_glDeleteFramebuffers;
+  if (!strcmp(name, "glDeleteRenderbuffers")) return emscripten_glDeleteRenderbuffers;
+  if (!strcmp(name, "glFramebufferTexture1D")) return emscripten_glFramebufferTexture1D;
+  if (!strcmp(name, "glFramebufferTexture2D")) return emscripten_glFramebufferTexture2D;
+  if (!strcmp(name, "glFramebufferTexture3D")) return emscripten_glFramebufferTexture3D;
+  if (!strcmp(name, "glTexImage2D")) return emscripten_glTexImage2D;
+  // If LEGACY_GL_EMULATION is on, allow access to those functions too
+  if (EM_ASM_INT({ return GL.legacyGLEmulation })) {
+    if (!strcmp(name, "glGetShaderPrecisionFormat")) return emscripten_glGetShaderPrecisionFormat;
```

Comment:
Wouldn't it be better for DCE if this check could be done by the processor?    (I guess then we would need to two different version of libGL though :(

Reply:
Yeah, that's the exact issue.

In general people using this code path are already getting a large amount of undce-able code, so it seems low priority to optimize, and not worth system additional library complexity, I think.